tests/commands/package/test_call.py

```diff
@@ -758,3 +762,141 @@ def test_already_packaged(package_command, first_app, tmp_path):
     # package_app() call, no new artefact is created; the absence
     # of this file shows that the old one has been deleted.
     assert not artefact_path.exists()
+
+
+def test_package_app_single(package_command, first_app):
+    """If the --app flag is used, only the selected app is packaged."""
+    # Add a single app
+    package_command.apps = {
+        "first": first_app,
+    }
+
+    # Configure the --app option
+    options, _ = package_command.parse_options(["--app", "first"])
+
+    # Run the package command
+    package_command(**options)
+
+    # The right sequence of things will be done
+    assert package_command.actions == [
+        # Host OS is verified
+        ("verify-host",),
+        # Tools are verified
+        ("verify-tools",),
+        # App config has been finalized
+        ("finalize-app-config", "first"),
+        # App template is verified
+        ("verify-app-template", "first"),
+        # App tools are verified
+        ("verify-app-tools", "first"),
+        # Package the app
+        (
+            "package",
+            "first",
+            {
+                "adhoc_sign": False,
+                "identity": None,
+            },
+        ),
+    ]
+
+
+def test_package_app_invalid(package_command, first_app, second_app):
+    """If an invalid app name is passed to --app, raise an error."""
+    # Add two apps
+    package_command.apps = {
+        "first": first_app,
+        "second": second_app,
+    }
+
+    # Configure the --app option with an invalid app
+    options, _ = package_command.parse_options(["--app", "invalid"])
+
+    # Running the command should raise an error
+    with pytest.raises(
+        BriefcaseCommandError,
+        match=r"App 'invalid' does not exist in this project.",
+    ):
+        package_command(**options)
+
+
+def test_package_app_none_defined(package_command):
+    """Packaging when no apps are defined should be a no-op."""
+    # No apps defined
+    package_command.apps = {}
+
+    # Configure no command line options
+    options, _ = package_command.parse_options([])
+
+    # Run the package command
+    result = package_command(**options)
+
+    # The right sequence of things will be done
+    assert result is None
+    assert package_command.actions == [
+        # Host OS is verified
+        ("verify-host",),
+        # Tools are verified
+        ("verify-tools",),
+    ]
+
+
+def test_package_with_update(package_command, first_app, tmp_path):
```

Comment:
This is lot closer to the level of testing needed here - but we should also be testing the `-a first` version of the flag.